### PR TITLE
Wire up CSV uploads, sync the new table

### PR DESCRIPTION
**Pull request by tsmacdonald**
Still need to create a model (next PR)

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/29856)

### CODE REVIEW

File: test/metabase/csv_test.clj [hunk 1]
```diff
@@ -127,3 +130,71 @@
...
+                                             uploads-schema-name  "public"
+                                             uploads-table-prefix "uploaded_magic_"]
+            (csv/load! file)
+            (let [new-table (t2/select-one Table {:where
+                                                  [:and [:= :db_id db-id]
+                                                        (when (seq existing-table-ids)
+                                                          [:not-in :id existing-table-ids])]})]
+              (is (str/starts-with? (:name new-table) "pokefans"))
+              (is (= "public" (:schema new-table)))
+              (is (= #{"id" "name"} (t2/select-fn-set :name Field :table_id (:id new-table)))))))))))
```
Comment (tsmacdonald) [inline test/metabase/csv_test.clj:200]: I figured all of the tricky cases are handled by the tests for the component functions—the point of this `deftest` is to make sure everything is coordinated right.

**Comment by tsmacdonald** [comment]
[Fixes #29759]

**Review by calherries — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29856)* status: 0 of 3 files reviewed, 3 unresolved discussions (waiting on @tsmacdonald)

___
*[`src/metabase/csv.clj` line 154 at r3](https://reviewable.io/reviews/metabase/metabase/29856#-NSiYOcm4fLRQKaR6zhv:-NSiYOcm4fLRQKaR6zhw:bn1o86v) ([raw file](https://github.com/metabase/metabase/blob/3a3ac207ed9fc7547e3af085b5654d09786c5f92/src/metabase/csv.clj#L154)):*
> ```Clojure
>   [csv-file]
>   (when (not (setting/get :uploads-enabled))
>     (throw (Exception. "Uploads are not enabled.")))
> ```

We need to check whether actions are enabled on the database too. The action and upload settings are not kept in sync.
___
*[`src/metabase/csv.clj` line 161 at r3](https://reviewable.io/reviews/metabase/metabase/29856#-NSiWr8E8dF-tSjXGeK1:-NSiWr8E8dF-tSjXGeK2:bnu5tlt) ([raw file](https://github.com/metabase/metabase/blob/3a3ac207ed9fc7547e3af085b5654d09786c5f92/src/metabase/csv.clj#L161)):*
> ```Clojure
>         _table-prefix (get-setting-or-throw! :uploads-table-prefix)
>         driver        (driver.u/database->driver database)
>         _             (or (driver/database-supports? driver :csv-uploads nil)
> ```

Use the database value. Also using `when-not` over `or` feels more natural since we're throwing away the return value of `driver/database-supports?`.



_Suggestion:_
```Clojure
        _             (when-not (driver/database-supports? driver :csv-uploads database)
```
___
*[`test/metabase/csv_test.clj` line 200 at r1](https://reviewable.io/reviews/metabase/metabase/29856#gh-1159518200:-NSiY5J_2mLZ9r3X2vQl:b-g8uu9k) ([raw file](https://github.com/metabase/metabase/blob/2c5689b2a9d9bba1e87dff736ea92e595bce9722/test/metabase/csv_test.clj#L200)):*
<details><summary><i>Previously, tsmacdonald (Tim Macdonald) wrote…</i></summary><blockquote>

I figured all of the tricky cases are handled by the tests for the component functions—the point of this `deftest` is to make sure everything is coordinated right.
</blockquote></details>

We need to check whether actions are enabled on the database too.

**Review by calherries — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29856)* status: 0 of 3 files reviewed, 6 unresolved discussions (waiting on @tsmacdonald)

___
*[`src/metabase/csv.clj` line 119 at r3](https://reviewable.io/reviews/metabase/metabase/29856#-NSignbh-H3rpD1638Yi:-NSignbh-H3rpD1638Yj:b7kz3qj) ([raw file](https://github.com/metabase/metabase/blob/3a3ac207ed9fc7547e3af085b5654d09786c5f92/src/metabase/csv.clj#L119)):*
> ```Clojure
>   [setting-name]
>   (value-or-throw! (setting/get setting-name)
>                    (trs "You must set the `{0}` before uploading files." (name setting-name))))
> ```

The error message is for the user of the csv upload feature



_Suggestion:_
```Clojure
                   (tru "You must set the `{0}` before uploading files." (name setting-name))))
```
___
*[`src/metabase/csv.clj` line 157 at r3](https://reviewable.io/reviews/metabase/metabase/29856#-NSih2uTAJAx7g9gFjH3:-NSih2uTAJAx7g9gFjH4:undefined) ([raw file](https://github.com/metabase/metabase/blob/3a3ac207ed9fc7547e3af085b5654d09786c5f92/src/metabase/csv.clj#L157)):*
> ```Clojure
>   (let [db-id         (get-setting-or-throw! :uploads-database-id)
>         database      (or (t2/select-one Database :id db-id)
>                           (throw (Exception. (trs "The uploads database does not exist."))))
> ```





_Suggestion:_
```Clojure
                          (throw (Exception. (tru "The uploads database does not exist."))))
```
___
*[`src/metabase/csv.clj` line 162 at r3](https://reviewable.io/reviews/metabase/metabase/29856#-NSih6aS3o8g274703y-:-NSih6aS3o8g274703y0:undefined) ([raw file](https://github.com/metabase/metabase/blob/3a3ac207ed9fc7547e3af085b5654d09786c5f92/src/metabase/csv.clj#L162)):*
> ```Clojure
>         driver        (driver.u/database->driver database)
>         _             (or (driver/database-supports? driver :csv-uploads nil)
>                           (throw (Exception. (trs "Uploads are not supported on {0} databases." (str/capitalize (name driver))))))]
> ```





_Suggestion:_
```Clojure
                          (throw (Exception. (tru "Uploads are not supported on {0} databases." (str/capitalize (name driver))))))]
```

**Comment by tsmacdonald** [comment]
Closing in favor of #29966

**Pull request closed without merging**